### PR TITLE
Reduce the number of travis matrices

### PR DESCRIPTION
**Pull request by csordasmarton**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -124,15 +84,7 @@ addons:
...
-        if [[ "$TRAVIS_OS_NAME" = "linux" ]]; then
-          make -C analyzer test_build_logger
-        fi
-      elif [[ ! -z "$DATABASE" ]]; then
-        make -C web test_matrix_${DATABASE};
-      else
-        make test;
-      fi
+    - make -C analyzer test_unit test_functional
+    - if [[ "$TRAVIS_OS_NAME" = "linux" ]]; then make -C analyzer test_tu_collector test_build_logger; fi
```
Comment (gyorb) [inline .travis.yml]: does the `tu_collector` not work on osx? It should collect for clang compilation commands the required files too.

Comment (csordasmarton) [reply .travis.yml]: For some reason it doesn't work on osx. I tried to debug it in #1969 but unfortunately without success. I will give it a try later.

Comment (gyorb) [reply .travis.yml]: Ok, I've created a separate issue to fix it #1984

File: .travis.yml [hunk 2]
```diff
@@ -7,56 +7,16 @@ services:
 matrix:
   include:
     - os: linux
-      name: "Analyzer test cases"
+      name: "Analyzer and web test cases"
```
Comment (gyorb) [inline .travis.yml]: Please add the same name for linux too `"Analyzer and web test cases (sqlite, psycopg2, pg8000)"`

**Pull request merged**